server/neptune/workflows/internal/activities/github.go

```diff
@@ -204,3 +204,40 @@ func (a *githubActivities) FetchRoot(ctx context.Context, request FetchRootReque
 		LocalRoot: localRoot,
 	}, nil
 }
+
+type CompareCommitRequest struct {
+	DeployRequestRevision  string
+	LatestDeployedRevision string
+	Repo                   internal.Repo
+}
+
+type CompareCommitResponse struct {
+	DeployRequestRevisionAheadBy int
+}
+
+func (a *githubActivities) CompareCommit(ctx context.Context, request CompareCommitRequest) (CompareCommitResponse, error) {
+	client, err := a.ClientCreator.NewInstallationClient(request.Repo.Credentials.InstallationToken)
+	if err != nil {
+		return CompareCommitResponse{}, errors.Wrap(err, "creating installation client")
+	}
+
+	comparison, resp, err := client.Repositories.CompareCommits(ctx, request.Repo.Owner, request.Repo.Name, request.LatestDeployedRevision, request.DeployRequestRevision, &github.ListOptions{})
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return CompareCommitResponse{}, errors.Wrap(err, "comparing commits")
+	}
+
+	if comparison.GetStatus() == "" {
+		return CompareCommitResponse{}, errors.New("nil commit comparison status")
+	}
+
+	// aheadBy is 0 when behindBy gt 0. We subtract to give a negative value which indicates the deploy request revision is behind
```

Comment:
Thoughts on using a signed int that indicates if a commit is ahead, behind or identical? 

One key benefit of returning `aheadBy` count is we can use that number if > 1 to update other commits in between with the message 
```
This revision has already been deployed in this deployment ... 
``` 

Or a more cleaner way would be to have a CommitComparison object that has all of that under the hood!